src/nanovg.c

```diff
@@ -2119,22 +2119,30 @@ void nvgRect(NVGcontext* ctx, float x, float y, float w, float h)
 
 void nvgRoundedRect(NVGcontext* ctx, float x, float y, float w, float h, float r)
 {
-	if (r < 0.1f) {
-		nvgRect(ctx, x,y,w,h);
+	nvgRoundedRectEx(ctx, x, y, w, h, r, r, r, r);
+}
+
+void nvgRoundedRectEx(NVGcontext* ctx, float x, float y, float w, float h, float r1, float r2, float r3, float r4)
+{
+	if(r1 < 0.1f && r2 < 0.1f && r3 < 0.1f && r4 < 0.1f) {
+		nvgRect(ctx, x, y, w, h);
 		return;
 	}
 	else {
-		float rx = nvg__minf(r, nvg__absf(w)*0.5f) * nvg__signf(w), ry = nvg__minf(r, nvg__absf(h)*0.5f) * nvg__signf(h);
+		float r1x = nvg__minf(r1, nvg__absf(w)*0.5f) * nvg__signf(w), r1y = nvg__minf(r1, nvg__absf(h)*0.5f) * nvg__signf(h);
+		float r2x = nvg__minf(r2, nvg__absf(w)*0.5f) * nvg__signf(w), r2y = nvg__minf(r2, nvg__absf(h)*0.5f) * nvg__signf(h);
+		float r3x = nvg__minf(r3, nvg__absf(w)*0.5f) * nvg__signf(w), r3y = nvg__minf(r3, nvg__absf(h)*0.5f) * nvg__signf(h);
+		float r4x = nvg__minf(r4, nvg__absf(w)*0.5f) * nvg__signf(w), r4y = nvg__minf(r4, nvg__absf(h)*0.5f) * nvg__signf(h);
```

Comment:
I thought about it and well, I just wouldn't prefer having an array as a parameter because that would probably mean I can't use this function without another line of code to declare that array. That also means iterating to setup the rx and ry values isn't very doable.  I'm happy with it as is. Maybe if this were C++ there would absolutely be much more possibilities. I could replace those operations with a function, but I would still be left with the same lines of code with maybe an inch of columns saved. In the end would just be about reducing code rather than optimization of actual execution.